File: swift-bridge/Sources/CoreAnimationBridge/CATextLayer.swift
import Foundation
import QuartzCore

@_cdecl("ca_text_layer_get_wrapped")
public func ca_text_layer_get_wrapped(_ handle: UnsafeMutableRawPointer?) -> Bool {
    guard let layer: CATextLayer = caBorrow(handle) else { return false }
    return layer.isWrapped
}

@_cdecl("ca_text_layer_set_wrapped")
public func ca_text_layer_set_wrapped(_ handle: UnsafeMutableRawPointer?, _ value: Bool) {
    guard let layer: CATextLayer = caBorrow(handle) else { return }
    layer.isWrapped = value
}

@_cdecl("ca_text_layer_get_allows_font_subpixel_quantization")
public func ca_text_layer_get_allows_font_subpixel_quantization(_ handle: UnsafeMutableRawPointer?) -> Bool {
    guard let layer: CATextLayer = caBorrow(handle) else { return false }
    return layer.allowsFontSubpixelQuantization
}

@_cdecl("ca_text_layer_set_allows_font_subpixel_quantization")
public func ca_text_layer_set_allows_font_subpixel_quantization(_ handle: UnsafeMutableRawPointer?, _ value: Bool) {
    guard let layer: CATextLayer = caBorrow(handle) else { return }
    layer.allowsFontSubpixelQuantization = value
}

File: src/ca_text_layer.rs
use crate::layer::{LayerLike, TextLayer};

impl TextLayer {
    #[must_use]
    pub fn is_wrapped(&self) -> bool {
        unsafe { crate::ffi::ca_text_layer_get_wrapped(self.as_layer_ptr()) }
    }

    pub fn set_wrapped(&self, value: bool) {
        unsafe { crate::ffi::ca_text_layer_set_wrapped(self.as_layer_ptr(), value) };
    }

    #[must_use]
    pub fn allows_font_subpixel_quantization(&self) -> bool {
        unsafe {
            crate::ffi::ca_text_layer_get_allows_font_subpixel_quantization(self.as_layer_ptr())
        }
    }

    pub fn set_allows_font_subpixel_quantization(&self, value: bool) {
        unsafe {
            crate::ffi::ca_text_layer_set_allows_font_subpixel_quantization(
                self.as_layer_ptr(),
                value,
            )
        };
    }
}
